### DISCUSSION TITLE
Can I import a file to reader using external sources like S3 or FTP

### DISCUSSION THREAD

**Discussion by stephen-silveira** [PHPOffice/PhpSpreadsheet#3136] [Q&A] [answered]
Hello, I can't find a way to give a s3 Amazon link to the reader, it only accepts Filename and directly take the files from public server. 
There's a way to do this? I'm asking cause I use Heroku + S3 and heroku automatically purges and re redeploy all times, so I would be loosing files. 

Any suggestions or comments ?! I'm already like 3 days in a row in this problem.  

Thanks a lot.

**Comment by oleibman** [comment, marked as answer]
PR #2326 made a change that allowed writing to Amazon S3. It was needed because the parameters for fopen needed to be different for S3 than for other sources. I don't think distinguishing between S3 and other systems would be necessary for read, and I don't see why S3 read wouldn't be allowed if S3 write is. I have no way to test. The discussion in that PR gives an example of the setup needed to write; can you adapt it for read?

**Reply by stephen-silveira** [reply]
hey @oleibman i could not find that Pull Request you are referring? thanks for the explanation until now.

**Reply by stephen-silveira** [reply]
https://github.com/PHPOffice/PhpSpreadsheet/pull/2326 this one?

**Reply by oleibman** [reply]
Yes, 2326, sorry about the typo.
